icefoganalytics/travel-authorization
# Issue-230: Travel Desk Rank Travel Options – Make Travel Request Process More Linear

Relates to:

- https://github.com/icefoganalytics/travel-authorization/issues/230

# Context

After a brief time away from the Travel Authorization project, I couldn't remember the process to complete a travel request. Since I'm the primary developer on this project, I concluded that the process is simply too hard to understand in its current state. This PR attempts to make the process more linear and self explanatory.

# Implementation

Rework travel authorization request in wizard pattern.

# Screenshots

Initial steps broken out for simpler logic and to make it easier to understand what you need to do
![image](https://github.com/user-attachments/assets/e2e49ac0-d6b8-4de7-8655-2029ec505e7f)

Trip details now a separate page
![image](https://github.com/user-attachments/assets/079329b9-df85-485f-b40b-9d7bfab89300)

Estimate generation now its own page, so non-linear page flow is now linear
![image](https://github.com/user-attachments/assets/87f4ce49-41ab-4fc4-8da0-a998b5806b2e)

Review and submit page
![image](https://github.com/user-attachments/assets/79dad68b-06f3-4f44-a45b-46654b178a61)

Submit page with validation failed
![image](https://github.com/user-attachments/assets/7deec9ad-c247-4591-afc8-60461e2f1e9f)

Review travel request page, with revert to draft option on "back"
![image](https://github.com/user-attachments/assets/5dbf2984-6e55-4d14-9a02-207d8e9d51f3)

Review estimate page
![image](https://github.com/user-attachments/assets/eefdc386-7adf-46dc-bcf0-27149a02848d)

Waiting for approval page, with option to jump back to review details or estimate
![image](https://github.com/user-attachments/assets/75a43c8e-32ba-476b-8dc9-2400002d4fbe)

Pretty much full steps list
![image](https://github.com/user-attachments/assets/f609aa12-4951-4ec1-a9f4-47899d17b219)

# Testing Instructions

1. Run the test suite via `dev test` (or `dev test_api`)
2. Boot the app via `dev up`
3. Log in to the app at http://localhost:8080
4. Click on the "New Travel Request" button at the bottom of the dashboard.
5. Follow the steps in the wizard.
6. Log in as the user your supervisor user, and approve the request.
7. Log back in as your traveler.
8. Check that you can now enter your travel details.
9. Log in as your travel desk user.
10. Using steps from https://github.com/icefoganalytics/travel-authorization/pull/237 try to enter some request options.
> This probably will not work. I need to rebuild the option generation code as it is very unpredictable.
